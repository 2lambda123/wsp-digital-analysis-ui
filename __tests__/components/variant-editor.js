/* global describe, it, expect, jest */

import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import React from 'react'

import VariantEditor from '../../lib/components/variant-editor'

describe('Component > VariantEditor', () => {
  it('renders correctly', () => {
    const props = {
      createVariant: jest.fn(),
      currentScenario: {},
      modifications: [],
      saveScenario: jest.fn(),
      scenarioName: 'Test Scenario',
      showVariant: jest.fn(),
      updateVariant: jest.fn(),
      variants: [],
      scenarioId: '1',
      projectId: '1'
    }

    // mount component
    const tree = mount(
      <VariantEditor
        {...props}
        />
    )
    expect(mountToJson(tree)).toMatchSnapshot()
    const noCalls = [
      'createVariant',
      'saveScenario',
      'showVariant',
      'updateVariant'
    ]
    noCalls.forEach((fn) => {
      expect(props[fn]).not.toBeCalled()
    })
  })
})
