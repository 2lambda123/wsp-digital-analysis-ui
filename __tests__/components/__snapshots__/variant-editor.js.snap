exports[`Component > VariantEditor renders correctly 1`] = `
<VariantEditor
  createVariant={[Function]}
  currentScenario={Object {}}
  modifications={Array []}
  projectId="1"
  saveScenario={[Function]}
  scenarioId="1"
  scenarioName="Test Scenario"
  showVariant={[Function]}
  updateVariant={[Function]}
  variants={Array []}>
  <div
    className="ModificationGroup">
    <div
      className="ModificationGroupTitle">
      <a
        className="pull-right"
        onClick={[Function]}>
        <pure(Icon)
          type="plus">
          <i
            className="fa fa-plus fa-fw " />
        </pure(Icon)>
      </a>
    </div>
    <form
      className="Variants form-inline" />
  </div>
</VariantEditor>
`;
